{"ast":null,"code":"var _class;\nimport { throwError } from 'rxjs';\nimport { tap, catchError } from 'rxjs/operators';\nimport * as i0 from \"@angular/core\";\nimport * as i1 from \"@angular/common/http\";\nexport class ClasseService {\n  constructor(http) {\n    this.http = http;\n    this.url = \"http://localhost:9000/api/classes\";\n  }\n  getClasse() {\n    return this.http.get(this.url).pipe(tap(classe => console.log(\"classe\", classe)), catchError(this.handleError));\n  }\n  ajClasse(name, prof) {\n    return this.http.post(this.url, {\n      name,\n      prof\n    }).pipe(catchError(this.handleError));\n  }\n  handleError(error) {\n    if (error.status === 0) {\n      // A client-side or network error occurred. Handle it accordingly.\n      console.error('An error occurred:', error.error);\n    } else {\n      // The backend returned an unsuccessful response code.\n      // The response body may contain clues as to what went wrong.\n      console.error(`Backend returned code ${error.status}, body was: `, error.error);\n    }\n    // Return an observable with a user-facing error message.\n    return throwError(() => new Error('Something bad happened; please try again later.'));\n  }\n}\n_class = ClasseService;\n_class.ɵfac = function ClasseService_Factory(t) {\n  return new (t || _class)(i0.ɵɵinject(i1.HttpClient));\n};\n_class.ɵprov = /*@__PURE__*/i0.ɵɵdefineInjectable({\n  token: _class,\n  factory: _class.ɵfac,\n  providedIn: 'root'\n});","map":{"version":3,"mappings":";AAEA,SAAqBA,UAAU,QAAQ,MAAM;AAC7C,SAASC,GAAG,EAAEC,UAAU,QAAQ,gBAAgB;;;AAMhD,OAAM,MAAOC,aAAa;EAExBC,YAAoBC,IAAe;IAAf,SAAI,GAAJA,IAAI;IADP,QAAG,GAAC,mCAAmC;EACjB;EAEhCC,SAAS;IAEd,OAAO,IAAI,CAACD,IAAI,CAACE,GAAG,CAAC,IAAI,CAACC,GAAG,CAAC,CAACC,IAAI,CAC/BR,GAAG,CAACS,MAAM,IAAEC,OAAO,CAACC,GAAG,CAAC,QAAQ,EAACF,MAAM,CAAC,CAAC,EACzCR,UAAU,CAAC,IAAI,CAACW,WAAW,CAAC,CAC/B;EACH;EAEOC,QAAQ,CAACC,IAAY,EAAEC,IAAQ;IACpC,OAAO,IAAI,CAACX,IAAI,CAACY,IAAI,CAAC,IAAI,CAACT,GAAG,EAAC;MAC7BO,IAAI;MAACC;KACN,CAAC,CAACP,IAAI,CACLP,UAAU,CAAC,IAAI,CAACW,WAAW,CAAC,CAC7B;EACH;EAEQA,WAAW,CAACK,KAAwB;IAC1C,IAAIA,KAAK,CAACC,MAAM,KAAK,CAAC,EAAE;MACtB;MACAR,OAAO,CAACO,KAAK,CAAC,oBAAoB,EAAEA,KAAK,CAACA,KAAK,CAAC;KACjD,MAAM;MACL;MACA;MACAP,OAAO,CAACO,KAAK,CACX,yBAAyBA,KAAK,CAACC,MAAM,cAAc,EAAED,KAAK,CAACA,KAAK,CAAC;;IAErE;IACA,OAAOlB,UAAU,CAAC,MAAM,IAAIoB,KAAK,CAAC,iDAAiD,CAAC,CAAC;EACvF;;SAhCWjB,aAAa;;mBAAbA,MAAa;AAAA;;SAAbA,MAAa;EAAAkB,SAAblB,MAAa;EAAAmB,YAFZ;AAAM","names":["throwError","tap","catchError","ClasseService","constructor","http","getClasse","get","url","pipe","classe","console","log","handleError","ajClasse","name","prof","post","error","status","Error","factory","providedIn"],"sourceRoot":"","sources":["C:\\Users\\Mehdi\\Desktop\\Angular\\my-dashboard\\src\\app\\classe.service.ts"],"sourcesContent":["import { Injectable } from '@angular/core';\nimport { HttpClient, HttpErrorResponse } from '@angular/common/http';\nimport { Observable, throwError } from 'rxjs';\nimport { tap, catchError } from 'rxjs/operators';\nimport { IUser } from './pages/user';\n\n@Injectable({\n  providedIn: 'root'\n})\nexport class ClasseService {\n  private readonly url=\"http://localhost:9000/api/classes\";\n  constructor(private http:HttpClient) { }\n\n  public getClasse():Observable<any>\n  {\n    return this.http.get(this.url).pipe(\n        tap(classe=>console.log(\"classe\",classe)),\n        catchError(this.handleError)\n    );\n  }\n\n  public ajClasse(name: String, prof:Str):Observable<any>{\n    return this.http.post(this.url,{\n      name,prof\n    }).pipe(\n      catchError(this.handleError)\n    )\n  }\n\n  private handleError(error: HttpErrorResponse) {\n    if (error.status === 0) {\n      // A client-side or network error occurred. Handle it accordingly.\n      console.error('An error occurred:', error.error);\n    } else {\n      // The backend returned an unsuccessful response code.\n      // The response body may contain clues as to what went wrong.\n      console.error(\n        `Backend returned code ${error.status}, body was: `, error.error);\n    }\n    // Return an observable with a user-facing error message.\n    return throwError(() => new Error('Something bad happened; please try again later.'));\n  }\n}\n"]},"metadata":{},"sourceType":"module","externalDependencies":[]}